core/capabilities/webapi/trigger.go

```diff
@@ -1,18 +1,279 @@
+	"A trigger to start workflow execution from a web api call",
+)
+
+type Input struct {
+}
+type Config struct {
+	AllowedSenders []string                 `toml:"allowedSenders"`
+	AllowedTopics  []string                 `toml:"allowedTopics"`
+	RateLimiter    common.RateLimiterConfig `toml:"rateLimiter"`
+	// RequiredParams is advisory to the web trigger message sender it is not enforced.
+	RequiredParams []string `toml:"requiredParams"`
+}
+
+type webapiTrigger struct {
+	allowedSenders map[string]bool
+	allowedTopics  map[string]bool
+	ch             chan<- capabilities.TriggerResponse
+	config         Config
+	rateLimiter    *common.RateLimiter
+}
+
+type triggerConnectorHandler struct {
+	services.StateMachine
+
+	capabilities.CapabilityInfo
+	capabilities.Validator[Config, Input, capabilities.TriggerResponse]
+	connector           connector.GatewayConnector
+	lggr                logger.Logger
+	mu                  sync.Mutex
+	registeredWorkflows map[string]webapiTrigger
+}
+
+var _ capabilities.TriggerCapability = (*triggerConnectorHandler)(nil)
+var _ services.Service = &triggerConnectorHandler{}
+
+func NewTrigger(config string, registry core.CapabilitiesRegistry, connector connector.GatewayConnector, lggr logger.Logger) (*triggerConnectorHandler, error) {
+	if connector == nil {
+		return nil, errors.New("missing connector")
+	}
+	handler := &triggerConnectorHandler{
+		Validator:           capabilities.NewValidator[Config, Input, capabilities.TriggerResponse](capabilities.ValidatorArgs{Info: webapiTriggerInfo}),
+		connector:           connector,
+		registeredWorkflows: map[string]webapiTrigger{},
+		lggr:                lggr.Named("WorkflowConnectorHandler"),
+	}
+
+	return handler, nil
+}
+
+// processTrigger iterates over each topic, checking against senders and rateLimits, then starting event processing and responding
+func (h *triggerConnectorHandler) processTrigger(ctx context.Context, gatewayID string, body *api.MessageBody, sender ethCommon.Address, payload webapicapabilities.TriggerRequestPayload) error {
+	// Pass on the payload with the expectation that it's in an acceptable format for the executor
+	wrappedPayload, err := values.WrapMap(payload)
+	if err != nil {
+		return fmt.Errorf("error wrapping payload %s", err)
+	}
+	topics := payload.Topics
+
+	// empty topics is error for V1
+	if len(topics) == 0 {
+		return fmt.Errorf("empty Workflow Topics")
+	}
+
+	// workflows that have matched topics
+	matchedWorkflows := 0
+	// workflows that have matched topic and passed all checks
+	fullyMatchedWorkflows := 0
+	for _, trigger := range h.registeredWorkflows {
+		for _, topic := range topics {
+			if trigger.allowedTopics[topic] {
+				matchedWorkflows++
+				if !trigger.allowedSenders[sender.String()] {
+					err = fmt.Errorf("unauthorized Sender %s, messageID %s", sender.String(), body.MessageId)
+					h.lggr.Debugw(err.Error())
+					continue
+				}
+				if !trigger.rateLimiter.Allow(body.Sender) {
```

Comment:
I don't like that with the current implementation this can drop some feedback to the user.

Scenario is: a request is sent that matches the topic on multiple workflows W1 and W2.
Then W1 doesn't start a run because of rate limiting, and W2 does.
The user would not know from the Gateway's response what happened, they just get a "success".
Even though they intended to start two runs with the single API request.

This is mitigated once observability is added, since the user could see in the dashboard what happened, but it would be the most complete to give feedback in the API response as well. 

I'm not going to block this PR over it, since I think it's a decent change to how the Gateway responds, but we should have a ticket tracking to handle this. Can just be a note in the observability ticket to revisit and decide if needed.